Foundation/Array/Unboxed.hs

```diff
@@ -1027,6 +1029,74 @@ reverse a
             | otherwise = primMbaWrite ma i (primAddrIndex ba (sizeAsOffset (endI - i))) >> loop (i+Offset 1)
 {-# SPECIALIZE [3] reverse :: UArray Word8 -> UArray Word8 #-}
 
+-- Finds where are the insertion points when we search for a `needle`
+-- within an `haystack`.
+indices :: PrimType ty => UArray ty -> UArray ty -> [Offset ty]
+indices ned hy =
+  case haystackLen < needleLen of
```

Comment:
I think indicies should be partial on empty arrays, otherwise it loops forever. That partiality should be explained in the doc comment.